apps/condo/domains/resident/tasks/sendResidentsNoAccountNotifications.js

```diff
@@ -0,0 +1,218 @@
+
+const { CONTEXT_FINISHED_STATUS } = require('@condo/domains/miniapp/constants')
+
+const { BillingIntegrationOrganizationContext, BillingProperty, BillingReceipt } = require('@condo/domains/billing/utils/serverSchema')
+
+const { BILLING_RECEIPT_AVAILABLE_NO_ACCOUNT_TYPE } = require('@condo/domains/notification/constants/constants')
+const { sendMessage } = require('@condo/domains/notification/utils/serverSchema')
+
+const { Property } = require('@condo/domains/property/utils/serverSchema')
+const { Resident, ServiceConsumer } = require('@condo/domains/resident/utils/serverSchema')
+
+const REDIS_LAST_DATE_KEY = 'LAST_SEND_RESIDENTS_NO_ACCOUNT_NOTIFICATION_CREATED_AT'
+
+const logger = pino({
+    name: 'send_residents_no_account_billing_receipt_added_notifications',
+    enabled: falsey(process.env.DISABLE_LOGGING),
+})
+
+const makeMessageKey = (period, propertyId, residentId) => `${period}.${propertyId}.${residentId}`
+
+/**
+ * Prepares data for sendMessage to resident on available billing receipt, then tries to send the message
+ * @param context
+ * @param resident
+ * @param period
+ * @returns {Promise<boolean>}
+ */
+const prepareAndSendNotification = async (context, resident, period) => {
+    if (isEmpty(resident)) throw new Error('resident is required to send message')
+    if (isEmpty(period)) throw new Error('period is required to send message')
+
+    // TODO(DOMA-3376): Detect locale by resident locale instead of organization country.
+    const country = get(resident, 'residentOrganization.country')
+    const locale = get(COUNTRIES, country || DEFAULT_LOCALE).locale
+    const notificationKey = makeMessageKey(period, resident.property.id, resident.id)
+
+    const data = {
+        residentId: resident.id,
+        userId: resident.user.id,
+        url: `${conf.SERVER_URL}/billing/receipts`,
+        propertyId: resident.property.id,
+        period,
+    }
+
+    const messageType = BILLING_RECEIPT_AVAILABLE_NO_ACCOUNT_TYPE
+    const messageData = {
+        lang: locale,
+        to: { user: { id: resident.user.id } },
+        type: messageType,
+        meta: { dv: 1, data },
+        sender: { dv: 1, fingerprint: 'send-residents-no-account-notifications' },
+        uniqKey: notificationKey,
+    }
+
+    try {
+        await sendMessage(context, messageData)
+    } catch (e) {
+        // TODO(DOMA-3343): handle error type
+        logger.info({ message: 'sendMessage attempt:', error: e, messageData })
+
+        return Promise.resolve(false)
+    }
+
+    return Promise.resolve(true)
+}
+
+/**
+ * Detects all properties that have new billing receipts,
+ * then detects all residents within that properties that haven't added account numbers yet,
+ * and tries to send notifications to that residents.
+ * @param billingContext
+ * @param receiptsWhere
+ * @returns {Promise<void>}
+ */
+const sendResidentsNoAccountNotificationsForContext = async (billingContext, receiptsWhere) => {
```

Comment:
this logic is too strange

Reply:
It's too strange to use billing context and receiptsWhere because the billingContext is already would be a part of receipts where

Reply:
fixed